### PR TITLE
docs(ic): updated backward compatibility and versioning according to latest Catena-X guidelines

### PR DESCRIPTION
**Pull request by markuskeidl**
Changes here include:

- Rework of non-funcational requirement for backward compatibility and versioning to reflect the changes in the overall Catena-X guidelines
- Removed deprecated content (e.g., no longer valid EDC asset properties for notifications, deprecated aspect models)

@mhellmeier We also need to remove the deprecatedc APIs for notifications (v1.1.0 and v1.2.0) once the updated versions are available.

### CODE REVIEW

**Review by nirschlk — COMMENTED** [review]
looks fine to me.

File: docs-kits/kits/Industry Core Kit/Software Development View/part_uniqueidpush.mdx [hunk 1]
```diff
@@ -134,17 +134,13 @@ For the EDC asset for receiving Unique ID Push notifications, the following conv
...
-      "asset:prop:notificationtype": "uniqueidpush",
-      "asset:prop:notificationmethod": "receive"
     },
     "privateProperties": {},
     "dataAddress": {}
   }
   ```
 
-  Properties "asset:prop:*" must still be added to be backward compatible with previous releases. Properties "dct:type" and "cx-common:version" are used to classify the asset and are explained in the [Digital Twin KIT](https://eclipse-tractusx.github.io/docs-kits/kits/Digital%20Twin%20Kit/Software%20Development%20View/Specification%20Digital%20Twin%20KIT#registration-at-edc) in more detail. 
+  Properties "dct:type" and "cx-common:version" are used to classify the asset and are explained in the [Digital Twin KIT](https://eclipse-tractusx.github.io/docs-kits/kits/Digital%20Twin%20Kit/Software%20Development%20View/Specification%20Digital%20Twin%20KIT#registration-at-edc) in more detail.
```
[the remark this answers is not in the document]
Comment (markuskeidl) [reply docs-kits/kits/Industry Core Kit/Software Development View/part_uniqueidpush.mdx:143]: The namespaces are shown in the example. That's why I thought it was obvious. You specifically mean in the text after the example, right. I could just use "https://purl.org/dc/terms/type" as name of the property and not use the namespace in the explanation.

**Comment by markuskeidl** [comment]
With the latest commit, I updated the API backward compatibility part as the version number should be specified without patch version.

**Review by arnoweiss — APPROVED** [review]

**Comment by mhellmeier** [comment]
> @mhellmeier We also need to remove the deprecatedc APIs for notifications (v1.1.0 and v1.2.0) once the updated versions are available.

I will do this in another branch and PR.

**Review by mhellmeier — APPROVED** [review]

**Review by mhellmeier — APPROVED** [review]

**Pull request merged**